MCR-1397 QueuedJob keep running on any exceptions

adlerre opened the pull request.

adlerre committed:
MCR-1397 QueuedJob keep running on any exceptions

mcradmin commented on mycore-base/src/main/java/org/mycore/services/queuedjob/MCRJobMaster.java:
```diff
@@ -178,74 +177,76 @@ protected void beforeExecute(Thread t, Runnable r) {
+                                } catch (InterruptedException e) {
+                                    LOGGER.error("Job thread was interrupted.", e);
                                 }
-                            } catch (InterruptedException e) {
```
Database Exception may result in to much logs to handle, rest here for a while, please ;-)

adlerre committed:
MCR-1397 QueuedJob catch PersistenceException and wait for try again

adlerre committed:
MCR-1397 QueuedJob improve job handling

- use criteria builder
- on poll check if job is already processing
- add some test

adlerre committed:
MCR-1397 QueuedJob fix use of EntityManager
